### PR TITLE
dfix 0.2.2 (new formula)

### PR DESCRIPTION
**Pull request by John-Colvin**
Dfix is a tool for auto-upgrading D source code that uses deprecated or otherwise "bad" syntax.

### CODE REVIEW

**Comment by John-Colvin** [comment]
I would have liked to have put the proper test in the ```test do``` block, but it requires test files that are in the source repository.

**Comment by mikemcquaid** [comment]
> I would have liked to have put the proper test in the test do block, but it requires test files that are in the source repository.

You can write one out with `Pathname.write` and `EOS.undent`.

**Comment by John-Colvin** [comment]
```make test``` also uses a hardcoded path to the built executable. A symbolic link would do the trick, but overall it seems easier just to leave it in ```install``` for now.

File: Library/Formula/dfix.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  depends_on "dmd" => :build
+
+  def install
+    system "make"
+    system "make", "test"
+    bin.install "bin/dfix"
+  end
+
+  test do
+    system "dfix", "--help"
```
Comment (bfontaine) [inline Library/Formula/dfix.rb]: Please use `"#{bin}/dfix"`.

File: Library/Formula/dfix.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      :revision => "ee55825b695fce94a337ab6a2009366372649d46"
+
+  head "https://github.com/Hackerpilot/dfix.git", :shallow => false
+
+  depends_on "dmd" => :build
+
+  def install
+    system "make"
+    system "make", "test"
+    bin.install "bin/dfix"
```
Comment (bfontaine) [inline Library/Formula/dfix.rb]: Is there a `make install` that we can use?

Comment (John-Colvin) [inline Library/Formula/dfix.rb]: no

File: Library/Formula/dfix.rb (hunk 1 above)
Comment (John-Colvin) [inline Library/Formula/dfix.rb]: why? We have to be in the right folder otherwise ```make``` would fail, no?

Comment (mikemcquaid) [inline Library/Formula/dfix.rb]: Also, as discussed earlier, please update this test to operate on an actual file you write out.

Comment (DomT4) [inline Library/Formula/dfix.rb]: `bin` is the direct path to the formula's Cellar bin, i.e. `/usr/local/Cellar/ducks/1.2.3/bin/`. We use the direct path to ensure we're not calling another tool with the same name somewhere else on the system, for one. A functionality test would be preferable here though, as Mike requests.

Comment (John-Colvin) [inline Library/Formula/dfix.rb]: Sorry for the noise, I was looking at the wrong line.

**Comment by John-Colvin** [comment]
@mikemcquaid 
> You can write one out with Pathname.write and EOS.undent.

The files I'm referring to are a bit long to have sitting as strings in the formula file: https://github.com/Hackerpilot/dfix/tree/master/test

**Comment by mikemcquaid** [comment]
@John-Colvin I actually think that's fine (even if they can't be shortened which would be good to try).

**Comment by John-Colvin** [comment]
@mikemcquaid The other problem is that the test files could be dependent on the version (stable vs devel vs head). I could `share.install` the text files and then read them from there?

**Comment by mikemcquaid** [comment]
> @mikemcquaid The other problem is that the test files could be dependent on the version (stable vs devel vs head). I could share.install the text files and then read them from there?

:+1:

**Comment by DomT4** [comment]
`pkgshare` maybe, just to be tidy.

**Comment by John-Colvin** [comment]
should be good to go now.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**